File: Estudos/M1P1-25.2/Ex2b.r
calcula_paralelo <- function(...){
    probs <- c(...) #ele combina (c) todos os pametros que a função recebe e comoloca no vetor probs
    return(1-prod(1-probs)) #faz o produto de (1-probs[i]), ou sjea, para todos os parametros.
}

calcula_total <- function(){
    par_esq <- calcula_paralelo(0.88,0.89)
    par_dir <- calcula_paralelo(0.91,0.89,0.88)
    serial <-0.97

    return(par_esq*par_dir*serial)
}

cat("Probabilidade do circuito funcionar: ", calcula_total(), "\n")

File: Estudos/M1P1-25.2/Ex2b.py
def calcula_paralelo(p1,p2,p3=0):
    if(p3==0):
        return 1-((1-p1)*(1-p2))
    else:
        return 1-((1-p1)*(1-p2)*(1-p3))

def calcula_total():
    par_esq = calcula_paralelo(0.88,0.89)
    par_dir = calcula_paralelo(0.91,0.89,0.88)
    serial = 0.97

    return par_esq*par_dir*serial

print("Probabilidade do circuito funcionar: ", calcula_total())
    